src/Store.h

```diff
@@ -191,10 +193,7 @@ class StoreEntry : public hash_link, public Packable
     bool hasMemStore() const { return mem_obj && mem_obj->memCache.index >= 0; }
 
     /// whether this entry can feed collapsed requests and only them
-    bool hittingRequiresCollapsing() const { return EBIT_TEST(flags, ENTRY_REQUIRES_COLLAPSING); }
-
-    /// allow or forbid collapsed requests feeding
-    void setCollapsingRequirement(const bool required);
+    bool hittingRequiresCollapsing() const { return mem_obj && (mem_obj->getReply()->pstate < Http::Message::psParsed); }
```

Comment:
If code sets StoreEntry to an internally-generated (usually, but not always, as the result of an error) response, what will pstate of that response be? IIRC, for examples, look for most replaceHttpReply() callers.